Physical_Engine/Query/Height.cs

```diff
@@ -0,0 +1,75 @@
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+using BH.oM.Physical.Elements;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Returns the height of a generic opening.")]
+        [Input("opening", "A generic Opening object.")]
+        [Output("height", "The total height of the generic opening.")]
+        public static double IHeight(this IOpening opening)
+        {
+            return Height(opening as dynamic);
+        }
+
+        /***************************************************/
+
+        [Description("Returns the height of a door object.")]
+        [Input("door", "A door object.")]
+        [Output("height", "The total height of the door object.")]
+        public static double Height(this Door door)
+        {
+            if(door == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot query the height of a null door.");
+                return 0;
+            }
+
+            return BH.Engine.Geometry.Query.Height((door.Location.IExternalEdges().FirstOrDefault()));
+        }
+
+        /***************************************************/
+        /****             Fallback Methods              ****/
+        /***************************************************/
+        
+        private static double Height(this object opening)
+        {
+            BH.Engine.Reflection.Compute.RecordError(String.Format("Height query for {0} type is not implemented.",opening.GetType()));
+            return 0;
+        }
```

Comment:
Think this gets cleaner if it is dealt with straight in the physical_Engine, without a roundtrip by extracting the geometry. Think something like the suggested below should solve it.

Have also added the Length quantity attribute that should be put on any output returning a length of something :).

```suggestion
using BH.oM.Geometry;
using BH.Engine.Geometry;
using BH.Engine.Spatial;
using BH.oM.Quantities.Attributes;

namespace BH.Engine.Physical
{
    public static partial class Query
    {
        /***************************************************/
        /****               Public Methods              ****/
        /***************************************************/

        [Description("Returns the height of a generic opening.")]
        [Input("opening", "A generic Opening object.")]
        [Output("height", "The total height of the generic opening.", typeof(Length))]
        public static double Height(this IOpening opening)
        {
            if (opening == null)
            {
                BH.Engine.Reflection.Compute.RecordError("Cannot query the height of a null opening.");
                return 0;
            }
            BoundingBox bBox = opening.IBounds();
            return (bBox.Max.Z - bBox.Min.Z);
        }
```